tests/anoph/test_plink_converter.py

```diff
@@ -0,0 +1,136 @@
+
+
+@pytest.fixture
+def af1_sim_api(af1_sim_fixture):
+    return PlinkConverter(
+        url=af1_sim_fixture.url,
+        config_path=_af1.CONFIG_PATH,
+        major_version_number=_af1.MAJOR_VERSION_NUMBER,
+        major_version_path=_af1.MAJOR_VERSION_PATH,
+        pre=False,
+        gff_gene_type="protein_coding_gene",
+        gff_gene_name_attribute="Note",
+        gff_default_attributes=("ID", "Parent", "Note", "description"),
+        default_site_mask="funestus",
+        results_cache=af1_sim_fixture.results_cache_path.as_posix(),
+        taxon_colors=_af1.TAXON_COLORS,
+    )
+
+
+# N.B., here we use pytest_cases to parametrize tests. Each
+# function whose name begins with "case_" defines a set of
+# inputs to the test functions. See the documentation for
+# pytest_cases for more information, e.g.:
+#
+# https://smarie.github.io/python-pytest-cases/#basic-usage
+#
+# We use this approach here because we want to use fixtures
+# as test parameters, which is otherwise hard to do with
+# pytest alone.
+
+
+def case_ag3_sim(ag3_sim_fixture, ag3_sim_api):
+    return ag3_sim_fixture, ag3_sim_api
+
+
+def case_af1_sim(af1_sim_fixture, af1_sim_api):
+    return af1_sim_fixture, af1_sim_api
+
+
+@parametrize_with_cases("fixture,api", cases=".")
+def test_plink_converter(fixture, api: PlinkConverter, tmp_path):
+    # Parameters for selecting input data, filtering, and converting.
+    all_sample_sets = api.sample_sets()["sample_set"].to_list()
+
+    data_params = dict(
+        region=random.choice(api.contigs),
+        sample_sets=random.sample(all_sample_sets, 2),
+        site_mask=random.choice((None,) + api.site_mask_ids),
+        min_minor_ac=1,
+        max_missing_an=1,
+        thin_offset=1,
+        random_seed=random.randint(1, 2000),
+    )
+
+    # Load a ds containing the randomly generated samples and regions to get the number of available snps to subset from.
+    ds = api.biallelic_snp_calls(
+        **data_params,
+    )
+
+    n_snps_available = ds.sizes["variants"]
+    n_snps = random.randint(1, n_snps_available)
+
+    # Define plink params.
+    plink_params = dict(results_dir=tmp_path, n_snps=n_snps, **data_params)
+
+    # Make the plink files.
+    api.biallelic_snps_to_plink(**plink_params)
+
+    # Test to see if bed, bim, fam output files exist.
+    file_path = f"{tmp_path}/{plink_params['region']}.{plink_params['n_snps']}.{plink_params['min_minor_ac']}.{plink_params['thin_offset']}.{plink_params['max_missing_an']}"
+
+    assert os.path.exists(f"{file_path}.bed")
+    assert os.path.exists(f"{file_path}.bim")
+    assert os.path.exists(f"{file_path}.fam")
+
+    # Read bed, bim, and fam files (bed_reader searches for the .bim and.fam files matching the prefix of the .bed file).
+    bed = bed_reader.open_bed(f"{file_path}.bed")
+
+    # Load a ds containing the same data exported to PLINK to test against.
+    ds_test = api.biallelic_snp_calls(
+        **data_params,
+        n_snps=n_snps,
+    )
```

Comment:
It is possible that the dataset could be slightly different from what is written out by the plink converter, because the plink converter also checks for and removes any rows with all identical genotype calls. But we may not encounter that with the test dataset, so perhaps OK to ignore.